### PR TITLE
Review dashboard updates

### PR DESCRIPTION
**Pull request by Raemon**
Changes to review dashboard:
– Added reviews to the default left-column, with ability to sort them
– added the Frontpage Widget (sans recommendations) to the default-left column
– streamlined the instructions
– added pingbacks to ReviewVotingExpandedPost
– added a subtitle to the page (intended to replace the "Show Instructions", providing an alternate button to click on that takes you back to the default view)
– replaced "Show Instructions" with "post count", because that seemed more important (and streamlining out unnecessary UI seemed important to manage page complexity)

image

image

Major outstanding TODO: make the right-column sort posts with at least one review to the top, which the author hasn't voted on yet, which seems like a reasonable way to give people a "todo list of things to vote on" that's a manageable length, and rewarding reviews more.

### CODE REVIEW

**Comment by Raemon** [comment]
(I wrote in the instructions "sorts reviewed-posts-to-the-top" so either need to fix the instruction or fix the sorting before merging this)

File: packages/lesswrong/components/review/ReviewVotingPage.tsx [hunk 1]
```diff
@@ -456,6 +509,8 @@ function getPostOrder(posts: Array<PostsList>, votes: Array<qualitativeVote|quad
       return [post, voteForPost, voteScore, i, randomPermutation[i]]
     })
     .sort(([post1, vote1, voteScore1, i1, permuted1], [post2, vote2, voteScore2, i2, permuted2]) => {
```
Comment (Raemon) [inline packages/lesswrong/components/review/ReviewVotingPage.tsx:511]: Could use a sanity check here. Intended behavior is that posts you haven't voted on with at least one review are sorted to the top, followed by posts you've voted on, followed by everything else.

**Comment by Raemon** [comment]
I've now updated the sorting to reflect the new instructions.

File: packages/lesswrong/components/ea-forum/EAHome.tsx [hunk 2]
```diff
@@ -31,8 +31,11 @@ const EAHome = () => {
       {shouldRenderEventBanner && <EventBanner />}
       
       <StickiedPosts />
-      
-      {reviewIsActive() && <FrontpageReviewWidget />}
+
+      {/* EA forum look here */}
```
Comment (Raemon) [inline packages/lesswrong/components/ea-forum/EAHome.tsx:35]: the reason for this is that on the ReviewVotingPage, I wanted to use the FrontpageReviewWidget, but the size of the column didn't precisely correspond to the SingleColumnSeciton that FrontpageReviewWidget normally wrapped itself in.

File: packages/lesswrong/components/review/ReviewVotingExpandedPost.tsx [hunk 3]
```diff
@@ -3,13 +3,10 @@ import { postGetPageUrl } from '../../lib/collections/posts/helpers';
 import { REVIEW_YEAR } from '../../lib/reviewUtils';
 import { registerComponent, Components } from '../../lib/vulcan-lib';
 import { postPageTitleStyles } from '../posts/PostsPage/PostsPageTitle';
-import { REVIEW_COMMENTS_VIEW } from './ReviewVotingPage';
 import { Link } from '../../lib/reactRouterWrapper';
 
 const styles = theme => ({
   root: {
```
Comment (Raemon) [inline packages/lesswrong/components/review/ReviewVotingExpandedPost.tsx]: this is moved over to a higher-level container

File: packages/lesswrong/components/review/ReviewVotingPage.tsx [hunk 4]
```diff
@@ -16,22 +16,23 @@ import CachedIcon from '@material-ui/icons/Cached';
...
 import { annualReviewAnnouncementPostPathSetting, annualReviewStart } from '../../lib/publicSettings';
 import moment from 'moment';
 import { forumTypeSetting } from '../../lib/instanceSettings';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import Card from '@material-ui/core/Card';
 
 const isEAForum = forumTypeSetting.get() === 'EAForum'
 
-const VOTING_VIEW = "reviewVoting" // unfortunately this can't just inhereit from REVIEW_YEAR. It needs to exactly match a view-type so that the type-check of the view can pass.
```
Comment (Raemon) [inline packages/lesswrong/components/review/ReviewVotingPage.tsx:26]: The original purpose of these variables was to make it easier to keep track of how we created new views each year (i.e. reviews2018, reviews2019), which is no longer necessary.

File: packages/lesswrong/components/posts/PostsPage/PostBodyPrefix.tsx [hunk 5]
```diff
@@ -53,8 +53,8 @@ const PostBodyPrefix = ({post, query, classes}: {
   const currentUser = useCurrentUser();
 
   return <>
-    {reviewIsActive() && canNominate(currentUser, post) && <div className={classes.reviewVoting}>
-      <ReviewVotingWidget post={post}/>
+    {reviewIsActive() && postEligibleForReview(post) && <div className={classes.reviewVoting}>
+      {canNominate(currentUser, post) && <ReviewVotingWidget post={post}/>}
       <ReviewPostButton post={post} year={REVIEW_YEAR+""} reviewMessage={<LWTooltip title={`Write up your thoughts on what was good about a post, how it could be improved, and how you think stands the tests of time as part of the broader ${forumTitleSetting.get()} conversation`} placement="bottom">
```
Comment (Raemon) [inline packages/lesswrong/components/posts/PostsPage/PostBodyPrefix.tsx:58]: All users are supposed to be able to review posts, makes this true

**Comment by jpaddison3** [comment]
We should hide the button on FrontpageReviewWidget that links to ReviewVotingPage when on ReviewVotingPage.

**Review by jpaddison3 — APPROVED** [review]
Approved Subject To Looking At Comments (ASTLAC)

Note especially the comment about hiding the button.

I've pushed a fix for the sort and some ea-specific updates mirroring yours.

File: packages/lesswrong/components/ea-forum/EAHome.tsx (hunk 2 above)
Comment (Raemon) [reply packages/lesswrong/components/ea-forum/EAHome.tsx:35]: SingleColumnSection includes a margin-bottom. I think it is plausible we should be replacing the architecture with a... _single_ singleColumnSection + dividers in between. But probably can wait till this comes up a second time.

File: packages/lesswrong/components/review/ReviewVotingPage.tsx [hunk 6]
```diff
@@ -456,6 +507,8 @@ function getPostOrder(posts: Array<PostsList>, votes: Array<qualitativeVote|quad
       return [post, voteForPost, voteScore, i, randomPermutation[i]]
     })
     .sort(([post1, vote1, voteScore1, i1, permuted1], [post2, vote2, voteScore2, i2, permuted2]) => {
+      if (post1.reviewCount > 0 && post2.reviewCount == 0 && !post1.currentUserReviewVote) return 1;
+      if (post2.reviewCount > 0 && post1.reviewCount == 0 && !post1.currentUserReviewVote) return -1;
       if (voteScore1 < voteScore2) return -1;
```
[the remark this answers is not in the document]
Comment (Raemon) [reply packages/lesswrong/components/review/ReviewVotingPage.tsx:524]: I think I agree with your disagreement, probably should check in with Habryka about whatever the original reasoning was. Leaving for now though.

File: packages/lesswrong/components/review/ReviewVotingPage.tsx (hunk 1 above)
Comment (Raemon) [reply packages/lesswrong/components/review/ReviewVotingPage.tsx:519]: Yeah agreed. The thing that was high friction here was that I'd need to go set up a whole suite of review-esque data on the testing db. (or, I guess, make it possible to do easy unit testing of components that I trusted to matter. The "trust to matter" there is doing a lot of work)

**Pull request merged**